### PR TITLE
{python3_02} Python3: fix regexp format

### PR DESCRIPTION
**Pull request by cjmayo**

### CODE REVIEW

File: linkcheck/htmlutil/linkparse.py [hunk 1]
```diff
@@ -82,11 +82,11 @@
...
 # matcher for <meta http-equiv=refresh> tags
-refresh_re = re.compile(ur"(?i)^\d+;\s*url=(?P<url>.+)$")
-_quoted_pat = ur"('[^']+'|\"[^\"]+\"|[^\)\s]+)"
-css_url_re = re.compile(ur"url\(\s*(?P<url>%s)\s*\)" % _quoted_pat)
+refresh_re = re.compile(r"(?i)^\d+;\s*url=(?P<url>.+)$")
+_quoted_pat = r"('[^']+'|\"[^\"]+\"|[^\)\s]+)"
+css_url_re = re.compile(r"url\(\s*(?P<url>%s)\s*\)" % _quoted_pat)
 swf_url_re = re.compile("(?i)%s" % urlutil.safe_url_pattern)
-c_comment_re = re.compile(ur"/\*.*?\*/", re.DOTALL)
+c_comment_re = re.compile(r"/\*.*?\*/", re.DOTALL)
```
Comment (mgedmin) [inline linkcheck/htmlutil/linkparse.py]: On Python 2 the type of the string passed to re.compile (unicode vs str) is significant.  I'm not quite sure what the implications are, but I worry about breaking things that might not be caught by the test suite.

Of course, on Python 3 the `ur"..."` literals are invalid syntax, but we could double the backslashes and use `u"..."` literals.

Comment (cjmayo) [reply linkcheck/htmlutil/linkparse.py]: Doubled all `\` except those escaping `"`.
Prefixed all the strings with `u`, including swf_url_re. This does seem the right thing to do in general and not just for re:
https://python-future.org/futurize.html#separating-text-from-bytes

Comment (mgedmin) [reply linkcheck/htmlutil/linkparse.py]: TBH I no longer trust my judgement - I tried to reproduce the kinds of problems I seemed to remember you can get from mixing up bytes vs unicode strings and regexps on Python 2.  I couldn't get any breakage.

Comment (cjmayo) [reply linkcheck/htmlutil/linkparse.py]: Is the conclusion to go back to the original patch with `r"` (in all instances)?

Comment (mgedmin) [reply linkcheck/htmlutil/linkparse.py]: Yeah, sorry!

Comment (cjmayo) [reply linkcheck/htmlutil/linkparse.py]: No problem. Just cherry-picked the original commit in.

**Pull request merged**